app/uk/gov/hmrc/fileupload/controllers/envelope.scala

```diff
@@ -47,21 +49,51 @@ object EnvelopeReport {
       expiryDate = envelope.expiryDate,
       status = Some(envelope.status.name),
       metadata = envelope.metadata,
+      constraints = envelope.constraints,
       destination = envelope.destination,
       application = envelope.application,
       files = fileReports
     )
   }
-
 }
 
 case class CreateEnvelopeRequest(callbackUrl: Option[String] = None,
                                  expiryDate: Option[DateTime] = None,
-                                 metadata: Option[JsObject] = None)
+                                 metadata: Option[JsObject] = None,
+                                 constraints: Option[EnvelopeConstraintsO] = None)
+
+case class EnvelopeConstraintsO(maxNumFiles: Option[Int] = None,
```

Comment:
Not sure if I understand what you mean but CreateEnvelopeRequest is a command, not an event. However this comment made me think that defaults could be also applied when validating the command rather then here, it doesn't make a huge difference probably though...